src/containers/Network/NodesTable.tsx

```diff
@@ -99,59 +106,200 @@ export const NodesTable: FC<{ nodes: NodeData[] }> = ({
 }) => {
   const nodes = unformattedNodes ? formatLedgerHistory(unformattedNodes) : null
   const ledgerRange = nodes && getLedgerRange(nodes)
+  const [sortedField, setSortedField] = useState(null)
+  const [sortOrder, setSortOrder] = useState(null)
+
+  const requestSort = (key) => {
+    let direction = 'desc'
+    if (sortedField === key && sortOrder === 'desc') {
+      direction = 'asc'
+    }
+    setSortOrder(direction)
+    setSortedField(key)
+  }
 
   const { t } = useTranslation()
   const renderNode = (node) => (
     <tr key={node.node_public_key}>
-      <td className="pubkey text-truncate">{node.node_public_key}</td>
-      <td className="ip text-truncate">{node.ip}</td>
-      <td className="state center">
+      <td className={getClassNamesFor('pubkey text-truncate', 'pubkey')}>
+        {node.node_public_key}
+      </td>
+      <td className={getClassNamesFor('ip text-truncate', 'ip')}>{node.ip}</td>
+      <td className={getClassNamesFor('state center', 'server_state')}>
         <span className={node.server_state}>{node.server_state}</span>
       </td>
-      <td className="version">{getVersion(node.version)}</td>
-      <td className="last-ledger">{renderLastLedger(node.validated_ledger)}</td>
-      <td className="uptime">{durationToHuman(node.uptime)}</td>
-      <td className="peers right">
+      <td className={getClassNamesFor('version', 'rippled_version')}>
+        {getVersion(node.version)}
+      </td>
+      <td className={getClassNamesFor('last-ledger', 'last_ledger')}>
+        {renderLastLedger(node.validated_ledger)}
+      </td>
+      <td className={getClassNamesFor('uptime', 'uptime')}>
+        {durationToHuman(node.uptime)}
+      </td>
+      <td className={getClassNamesFor('peers right', 'peers')}>
         {node.inbound_count + node.outbound_count}
       </td>
       <td className="in-out">
         <small>
           ({node.inbound_count}:{node.outbound_count})
         </small>
       </td>
-      <td className="ledgers">
+      <td className={getClassNamesFor('ledgers', 'ledger_history')}>
         {renderLedgerHistory(node.ledgers, ledgerRange)}
       </td>
-      <td className="quorum right">{node.quorum}</td>
-      <td className="load-factor right">
+      <td className={getClassNamesFor('quorum right', 'quorum')}>
+        {node.quorum}
+      </td>
+      <td className={getClassNamesFor('load-factor right', 'load')}>
         {node.load_factor && node.load_factor > 1
           ? node.load_factor.toFixed(2)
           : ''}
       </td>
-      <td className="latency right">
+      <td className={getClassNamesFor('latency right', 'latency')}>
         {node.io_latency_ms && node.io_latency_ms > 1}
       </td>
     </tr>
   )
 
+  const compareSemanticVersions = (
```

Comment:
Could you add tests for this function?

Reply:
tests have now been added that cover the compareSemanticVersions, via sorting the Version column.

Reply:
I would like to see this function tested separately and cover all cases. similar to this:
https://github.com/ripple/explorer/blob/1e808ad9187e27207a3f42b1b63ea00c05a01acd/src/containers/shared/test/utils.test.ts#L13-L33